internal/pkg/cache/cache.go

```diff
@@ -268,7 +294,11 @@ func New(cfg Config) (h *Handle, err error) {
 	}
 
 	// Initialize the root directory of the cache
-	rootDir := path.Join(parentDir, SubDirName)
+	// https://github.com/apptainer/apptainer/issues/886
```

Comment:
I don't think it's helpful to refer to the issue here.  I think it's pretty clear that it is adding the architecture subdirectory if defined.